### PR TITLE
Add endpoint_type and vpc_endpoint_id variables and identifiers to enable private transfer

### PR DESCRIPTION
**Pull request by stefanwb**

### CODE REVIEW

**Review by marwinbaumannsbp — DISMISSED** [review]

File: variables.tf [hunk 1]
```diff
@@ -3,9 +3,16 @@ variable "name" {
...
+variable "endpoint_type" {
+  type        = string
+  description = "The endpoint type, can be VPC_ENDPOINT or PUBLIC"
+  default     = "PUBLIC"
+}
+
+variable "vpc_endpoint_id" {
+  type        = string
+  description = "The endpoint ID"
+  default     = ""
```
Comment (shoekstra) [inline variables.tf]: Can we use `null` instead of `""` for empty defaults please

File: main.tf [hunk 2]
```diff
@@ -35,7 +35,12 @@ resource "aws_transfer_server" "default" {
   identity_provider_type = "SERVICE_MANAGED"
   logging_role           = aws_iam_role.default.arn
   tags                   = var.tags
-  depends_on             = [aws_iam_role.default]
+  endpoint_type          = var.endpoint_type
+
+  endpoint_details {
+    vpc_endpoint_id = var.vpc_endpoint_id
+  }
+  depends_on = [aws_iam_role.default]
```
Comment (shoekstra) [inline main.tf]: Can we move this line above `endpoint_details` so it's with the other fields please?

`depends_upon` should be the last field in the group of fields.

**Review by shoekstra — CHANGES_REQUESTED** [review]

**Review by marwinbaumannsbp — APPROVED** [review]
Included in new commit

**Review by marwinbaumannsbp — DISMISSED** [review]
included changes

**Review by shoekstra — APPROVED** [review]

**Pull request merged**